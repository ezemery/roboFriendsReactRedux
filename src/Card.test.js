import React from 'react';
import { shallow } from 'enzyme';
import Card from './components/Card';



it("renders a component", ()=>{
    const wrapper = shallow(<Card name={"Eze"} email={"emereuwaonueze@gmail.com"} id={1}/>);
    expect(wrapper).toMatchSnapshot();
});

it("renders one component", ()=>{
    const wrapper = shallow(<Card />);
    expect(wrapper.length).toEqual(1);
});

