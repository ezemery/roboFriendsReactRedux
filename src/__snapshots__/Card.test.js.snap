// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    email="emereuwaonueze@gmail.com"
    id={1}
    name="Eze"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robots"
          src="https://robohash.org/1?size=200x200"
        />,
        <div>
          <h2>
            Eze
          </h2>
          <p>
            emereuwaonueze@gmail.com
          </p>
        </div>,
      ],
      "className": "tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robots",
          "src": "https://robohash.org/1?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Eze
            </h2>,
            <p>
              emereuwaonueze@gmail.com
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Eze",
            },
            "ref": null,
            "rendered": "Eze",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "emereuwaonueze@gmail.com",
            },
            "ref": null,
            "rendered": "emereuwaonueze@gmail.com",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robots"
            src="https://robohash.org/1?size=200x200"
          />,
          <div>
            <h2>
              Eze
            </h2>
            <p>
              emereuwaonueze@gmail.com
            </p>
          </div>,
        ],
        "className": "tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robots",
            "src": "https://robohash.org/1?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Eze
              </h2>,
              <p>
                emereuwaonueze@gmail.com
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Eze",
              },
              "ref": null,
              "rendered": "Eze",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "emereuwaonueze@gmail.com",
              },
              "ref": null,
              "rendered": "emereuwaonueze@gmail.com",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
